test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1204,17 +1206,7 @@ public static SpatialPoint randomGeoPoint() {
     public static SpatialPoint randomCartesianPoint() {
         double x = randomDoubleBetween(-Float.MAX_VALUE, Float.MAX_VALUE, true);
         double y = randomDoubleBetween(-Float.MAX_VALUE, Float.MAX_VALUE, true);
-        return new SpatialPoint() {
-            @Override
-            public double getX() {
-                return x;
-            }
-
-            @Override
-            public double getY() {
-                return y;
-            }
-        };
+        return new SpatialPoint(x, y);
```

Comment:
Oh this is nicer!

Reply:
This is unfortunately reverted to keep SpatialPoint as an interface. Getting is back to a concrete class will require more work, as the current attempt was seen as an awkward compromise.